### PR TITLE
Add specific exceptions for import relationships errors

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Rename Mute to SuppressOutput, add possibility to check output messages in SuppressOutput rule.
Cleanup javadoc.

### CODE REVIEW

File: community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java [hunk 1]
```diff
@@ -645,4 +655,35 @@ static void warn( String warning )
...
+
+        public String getReference(Anchor anchor)
+        {
+            return page + "#" + anchor.anchor;
+        }
+    }
+
+    private enum Anchor
+    {
+        ID_SPACES("_id_spaces"),
```
Comment (nawroth) [inline community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: The leading underscore means the anchor id was generated from the heading, which might get changed.
It's better to set an id on the target like so: https://github.com/nawroth/neo4j/commit/c1a85236c8e0dfb21d956f175d1934eb78c2a89a
and use that instead.

File: community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java [hunk 2]
```diff
@@ -619,6 +620,60 @@ public void shouldDisallowMultilineFieldsByDefault() throws Exception
...
+                "/import-tool-header-format.html#import-tool-header-format-rels" );
+        shouldPrintReferenceLinkAsPartOfErrorMessage(
+                nodeIds(),
+                IteratorUtil.iterator( new RelationshipDataLine( "1", "2", "", "name" ) ),
+                "Relationship missing mandatory field 'TYPE', read more about relationship " +
+                "format in the manual:  http://neo4j" +
+                ".com/docs/" + Version.getKernelVersion() +
+                "/import-tool-header-format.html#import-tool-header-format-rels" );
+        shouldPrintReferenceLinkAsPartOfErrorMessage(
+                Arrays.asList( "1", "1"),
```
Comment (chrisvest) [inline community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java]: There are some places where the formatting isn't quite right.

File: community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java [hunk 3]
```diff
@@ -619,6 +620,60 @@ public void shouldDisallowMultilineFieldsByDefault() throws Exception
...
+                IteratorUtil.iterator( new RelationshipDataLine( "1", "2", "type", "name" ) ),
+                "Duplicate input ids that would otherwise clash can be put into separate id space, read more " +
+                "about how to use id spaces in the manual: http://neo4j.com/docs/" + Version.getKernelVersion() +
+                "/import-tool-header-format.html#_id_spaces" );
+    }
+
+    private void shouldPrintReferenceLinkAsPartOfErrorMessage(List<String> nodeIds, Iterator<RelationshipDataLine>
+            relationshipDataLines, String message) throws Exception
```
Comment (chrisvest) [inline community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java]: Like here. Don't put line breaks between type and identifier.

**Comment by chrisvest** [comment]
Next time it'd be helpful if the formatting changes and the functional changes were in different commits. Otherwise nice.

File: community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java (hunk 1 above)
Comment (MishaDemianenko) [inline community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: Nice thx, will do

**Comment by MishaDemianenko** [comment]
@nawroth id changed to proposed;
@chrisvest formatting updated, still not used to code style and sometimes idea just do it wrong :(

**Comment by chrisvest** [comment]
@MishaDemianenko this needs a rebase.

**Comment by chrisvest** [comment]
Looks good to me, though.

**Comment by MishaDemianenko** [comment]
@chrisvest rebased

**Pull request merged**